tensorflow/core/kernels/training_ops.cc

```diff
@@ -3415,6 +3415,9 @@ class SparseApplyKerasMomentumOp : public OpKernel {
                                 accum.shape().DebugString()));
     OP_REQUIRES(ctx, TensorShapeUtils::IsVectorOrHigher(var.shape()),
                 errors::InvalidArgument("var must be at least 1 dimensional"));
+    if (var.dtype() != DT_FLOAT || accum.dtype() != DT_FLOAT) {
```

Comment:
I'm not sure this restriction is comprehensive enough. Below this kernel is defined for all float types and complex types via the `TF_CALL_FLOAT_TYPES` and `TF_CALL_COMPLEX_TYPES` macros.

Is this change needed to fix anything in particular? If so can we add a test that demonstrates what is broken before and how this change might fix that?

Reply:
> I'm not sure this restriction is comprehensive enough. Below this kernel is defined for all float types and complex types via the `TF_CALL_FLOAT_TYPES` and `TF_CALL_COMPLEX_TYPES` macros.

This Op has other args like `lr`, `grad` & `momentum` that are registered for dtype `T` (numbertype) and indices for `Tindices` ({int32, int64}) which the template is designed for may be?.

I am not sure what actually `resource` means here whether it can limit any dtypes etc.